akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -2270,3 +2271,70 @@ private[akka] final class StatefulMapConcat[In, Out](val f: () => In => Iterable
   override def toString = "StatefulMapConcat"
 
 }
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] final case class AggregateWithBoundary[In, Agg, Out](
```

Comment:
refactored. see if it's ready for merge.

Reply:
You mentioned in some other comment I can't reply to, but I think it would be nice with a separate file for the operator, we have historically piled operators in this Ops file way past the point when it grew too large.